src/realm/table.hpp

```diff
@@ -902,10 +902,10 @@ class ColKeyIterator {
 
 private:
     friend class ColKeys;
-    const Table* m_table;
+    ConstTableRef m_table;
```

Comment:
I guess this is probably harmless, but using ColKeys as anything other than a temporary that doesn't need a strong reference to the table seems very suspicious and is probably broken even if it doesn't hit UB.

Reply:
it was caught by a few new asan configuraions. How it wasn't fixed in the test in 5 years i don't understand really. Now just it will at least check this situation with the ref.